client/src/store/likesList/reducers.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    GET_LIKE_LIST_PENDING,
+    GET_LIKE_LIST_SUCCESS,
+    GET_MORE_LIKE_LIST_SUCCESS,
+    UPLOAD_NEXT_LIKE_LIST,
+    ALL_LIKE_LIST_LOADED,
+    CLEAR_LIKE_LIST_LOADED,
+} from './actionTypes';
+import { IUser } from '../../components/FriendsRecommendations';
+
+export interface INewsFeed {
+    _id: string;
+    description: string;
+    comments: [];
+    tags: [];
+    authorsOfLike: [];
+    likeExist: boolean;
+    imgPath: string;
+    createdAt: string;
+    author: {
+        _id: string;
+        username: string;
+        photoPath: string;
+    };
+}
+
+export interface IFeedState {
+    feed: any;
+    page: number;
+    feedLoaded: boolean;
+    feedLoading: boolean;
+    friendsRecommendations: {users: IUser[], loading: boolean};
+}
+
+const defaultState = {
+    feed: [
+        {
+            description: '',
+            comments: [],
+            tags: [],
+            authorsOfLike: [],
+            likeExist: false,
+            _id: '',
+            imgPath: '',
+            author: {
+                photoPath: '',
+                _id: '',
+                username: '',
+            },
+            createdAt: '',
+        },
+    ],
+    page: 1,
+    feedLoaded: false,
+    feedLoading: false,
+    friendsRecommendations: {
+        users: [],
+        loading: false,
+    },
+};
+
+export const likeListReducer = (
+    state: IFeedState = defaultState,
+    action: { type: string, payload: any }): any => {
+    switch (action.type) {
+        case GET_LIKE_LIST_PENDING:
+            return {
+                ...state,
+                feedLoading: true,
+            };
+
+        case GET_LIKE_LIST_SUCCESS:
+            return {
+                ...state,
+                feed: action.payload,
+                feedLoading: false,
+            };
+
+        case GET_MORE_LIKE_LIST_SUCCESS:
```

Comment:
can we combine this and prev action?

Reply:
No, because GET_LIKE_LIST_SUCCESSis invoked when component mounted and it get data just for first page. GET_MORE_LIKE_LIST_SUCCESS is invoked by Waypoint and the current page is increasing.